dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +147,233 @@ async def get_cluster_status(self, req):
+        return aiohttp.web.Response(
+            text=WARNING_FOR_MULTI_TASK_IN_A_WORKER
+            + str(task_ids_in_a_worker)
+            + "\n"
+            + reply.output
+            if len(task_ids_in_a_worker) > 1
+            else reply.output
+        )
+
+    @routes.get("/task/cpu_profile")
+    async def get_task_cpu_profile(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the CPU profile for a specific task.
+        Note that one worker process works on one task at a time or one worker works on multiple async tasks.
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the CPU profile data.
+
+        Raises:
+            ValueError: If the "task_id" parameter is missing in the request query.
+            ValueError: If the "attempt_number" parameter is missing in the request query.
+            ValueError: If the maximum duration allowed is exceeded.
+            ValueError: If the worker begins working on another task during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is an internal server error during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If the CPU Flame Graph information for the task is not found.
+        """
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+        if "node_id" not in req.query:
+            raise ValueError("node_id is required")
+
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+        node_id = req.query.get("node_id")
+
+        ip = DataSource.node_id_to_ip[node_id]
+
+        try:
+            (pid, _) = await self.get_worker_details_for_task(task_id, attempt_number)
+        except ValueError as e:
+            raise aiohttp.web.HTTPInternalServerError(text=str(e))
+
+        duration = int(req.query.get("duration", 5))
+        if duration > 60:
+            raise ValueError(f"The max duration allowed is 60: {duration}.")
+        format = req.query.get("format", "flamegraph")
+
+        # Default not using `--native` for profiling
+        native = req.query.get("native", False) == "1"
+        reporter_stub = self._stubs[ip]
+
+        logger.info(
+            "Sending CPU profiling request to {}:{} for {} with native={}".format(
+                ip, pid, task_id, native
+            )
+        )
+
+        reply = await reporter_stub.CpuProfiling(
+            reporter_pb2.CpuProfilingRequest(
+                pid=pid, duration=duration, format=format, native=native
+            )
+        )
+
+        ## Call the API again to make sure task is still running and the worker is still working on the task
```

Comment:
Can you add a comment like: 
```
in order to really know for sure if there's any other task running during the profiling, we need to get all tasks 
that are running or finished, and parse the task events (i.e. when they start/finish) to see if there's overlap. It's 
too much to do so we do best efforts check here. 
```

Reply:
Sure! Btw, do we have a docs to explain the worker reuse strategy? Which means a worker may begin to work on other tasks once a task is finished, in order to maximize the efficiency of workers